doc/designs/prci_checker.md

```diff
@@ -0,0 +1,240 @@
+- `copr_defined`
+
+Let's take an example with `nightly_latest_389ds` definition:
+``` yaml
+nightly_latest_389ds:
+  job_prefix: '389ds-fedora/'
+  update_packages: True
+  copr_defined: True
+```
+
+The specifications shown above mean that every job defined in nightly_latest_389ds must:
+- Have a name starting with `389ds-fedora/`
+- Define `update_packages: True` and `copr_defined: True` <br>
+
+  > **Reason**: Since the file is testing with a copr repo for 389ds, the test must call dnf update  in order to update the package to the version provided in the copr repo. Otherwise they
+  would run with a wrong version of 389ds and would not properly test the integration). Likewise, if they define update_packages but do not set any copr repo, the wrong packages would be used.
+
+On top of the above requirements, job class should also be one of the supported (listed in prci_jobs_spec.yaml).
+
+For instance, the following would be a well formed job:
+``` yaml
+389ds-fedora/simple_replication:
+  requires: [389ds-fedora/build]
+  priority: 50
+  job:
+    class: RunPytest
+    args:
+      build_url: '{389ds-fedora/build_url}'
+      update_packages: True
+      copr: '@389ds/389-ds-base-nightly'
+      test_suite: test_integration/test_simple_replication.py
+      template: *ci-master-latest
+      timeout: 3600
+      topology: *master_1repl
+```
+
+
+## Error cases examples
+
+After adding new jobs to the definition or editing the content, it is common that a developer has made some mistakes following the convention. This checker will help us correcting these errors. Examples:
+
+###   Wrong or unsupported class:
+
+``` bash
+$ python3 ipatests/prci_definitions/prci_checker.py -f ipatests/prci_definitions/gating.yaml
+BEGINNING PRCI JOB DEFINITIONS CHECKS
+[File] gating [Jobs] 24
+ERROR: In job 'fedora-latest/test_installation_TestInstallMaster':
+  'class' field should be defined with one of the supported: ['RunPytest', 'RunPytest2', 'RunPytest3', 'RunWebuiTests', 'RunADTests']
+CHECKS FINISHED WITH ERRORS
+```
+
+### Wrong job prefix in gating:
+
+``` bash
+$ python3 ipatests/prci_definitions/prci_checker.py -f ipatests/prci_definitions/gating.yaml
+BEGINNING PRCI JOB DEFINITIONS CHECKS
+[File] gating [Jobs] 24
+ERROR: In job 'fedora-latdest/test_installation_TestInstallMaster':
+  Job name should start with prefix 'fedora-latest/'
+CHECKS FINISHED WITH ERRORS
+```
+
+### Wrong or missing parameter values.
+E.g.
+1. Missing `selinux_enforcing: True` in selinux job definition:
+
+``` bash
+$ python3 ipatests/prci_definitions/prci_checker.py -f ipatests/prci_definitions/nightly_latest_selinux.yaml
+BEGINNING PRCI JOB DEFINITIONS CHECKS
+[File] nightly_latest_selinux [Jobs] 149
+ERROR: In job 'fedora-latest/simple_replication':
+  Job field "selinux_enforcing" should be defined as: "selinux_enforcing: True"
+CHECKS FINISHED WITH ERRORS
+```
+
+2. Field `topology` is not defined with one of the pre-defined templates:
+``` bash
```

Comment:
Add an empty line before the code block, otherwise it is not properly displayed.